In [31]:
#Importing the required libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

# Data Dictionary

#### A1: Question 1 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used  

#### A2: Question 2 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used  

#### A3: Question 3 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used  

#### A4: Question 4 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used 

#### A5: Question 5 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used 

#### A6: A6: Question 6 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used 

#### A7: Question 7 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used  

#### A8: Question 8 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used 

#### A9: Question 9 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used  

#### A:10 Question 10 Answer 	Binary (0, 1)	The answer code of the question based on the screening method used  

#### Age	Number 	Toddlers (months)

#### Score by Q-chat-10 	Number	1-10 (Less that or equal 3 no ASD traits; > 3 ASD traits

#### Sex	Character	Male or Female 

#### Ethnicity	String	List of common ethnicities in text format 

#### Born with jaundice	Boolean  (yes or no)	Whether the case was born with jaundice

#### Family member with ASD history	Boolean  (yes or no)	Whether any immediate family member has a PDD 

#### Who is completing the test	String 	Parent, self, caregiver, medical staff, clinician ,etc.

#### Why_are_you_taken_the_screening	String	Use input textbox 

#### Class variable 	String 	ASD traits or No ASD traits (automatically assigned by the ASDTests app). (Yes / No)


# Importing data

In [32]:
df = pd.read_csv("Toddler Autism dataset July 2018.csv")
df.head()

Case_No  A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Age_Mons  Qchat-10-Score  \
0        1   0   0   0   0   0   0   1   1   0    1        28               3   
1        2   1   1   0   0   0   1   1   0   0    0        36               4   
2        3   1   0   0   0   0   0   1   1   0    1        36               4   
3        4   1   1   1   1   1   1   1   1   1    1        24              10   
4        5   1   1   0   1   1   1   1   1   1    1        20               9   

  Sex       Ethnicity Jaundice Family_mem_with_ASD Who completed the test  \
0   f  middle eastern      yes                  no          family member   
1   m  White European      yes                  no          family member   
2   m  middle eastern      yes                  no          family member   
3   m        Hispanic       no                  no          family member   
4   f  White European       no                 yes          family member   

  Class/ASD Traits   
0                No  
1               Yes  
2               Yes  
3               Yes  
4               Yes

In [33]:
# This code snippet uses the pandas library to read in a CSV file named "Toddler Autism dataset July 2018.csv"
# and assigns it to the variable "df"

import pandas as pd

df = pd.read_csv("Toddler Autism dataset July 2018.csv")

# The "head()" function is then called on the "df" DataFrame, which returns the first five rows of the data
# This is useful for quickly checking the structure and contents of the DataFrame
df.head()


Case_No  A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Age_Mons  Qchat-10-Score  \
0        1   0   0   0   0   0   0   1   1   0    1        28               3   
1        2   1   1   0   0   0   1   1   0   0    0        36               4   
2        3   1   0   0   0   0   0   1   1   0    1        36               4   
3        4   1   1   1   1   1   1   1   1   1    1        24              10   
4        5   1   1   0   1   1   1   1   1   1    1        20               9   

  Sex       Ethnicity Jaundice Family_mem_with_ASD Who completed the test  \
0   f  middle eastern      yes                  no          family member   
1   m  White European      yes                  no          family member   
2   m  middle eastern      yes                  no          family member   
3   m        Hispanic       no                  no          family member   
4   f  White European       no                 yes          family member   

  Class/ASD Traits   
0                No  
1               Yes  
2               Yes  
3               Yes  
4               Yes

In [34]:
# renaming the columns of the dataframe

df.columns = df.columns.str.replace('/','_').str.replace(' ','')

rename_columns = {
    'Age_Mons':'Age_In_Months',
    'Qchat-10-Score':'Qchat_10_Score',
    'Family_mem_with_ASD': 'Family_Mem_With_ASD',
    'Class/ASD Traits' : 'ASD_Diagnosed',
    'Whocompletedthetest' : 'Who_Completed_The_Test'
}

df.rename(columns=rename_columns, inplace=True)

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1054 entries, 0 to 1053
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Case_No                 1054 non-null   int64 
 1   A1                      1054 non-null   int64 
 2   A2                      1054 non-null   int64 
 3   A3                      1054 non-null   int64 
 4   A4                      1054 non-null   int64 
 5   A5                      1054 non-null   int64 
 6   A6                      1054 non-null   int64 
 7   A7                      1054 non-null   int64 
 8   A8                      1054 non-null   int64 
 9   A9                      1054 non-null   int64 
 10  A10                     1054 non-null   int64 
 11  Age_In_Months           1054 non-null   int64 
 12  Qchat_10_Score          1054 non-null   int64 
 13  Sex                     1054 non-null   object
 14  Ethnicity               1054 non-null   object
 15  Jaun

#### Inference

* No null values are present
* Total 1054 records of toddlers and 18 predictors excluding the target variable
* Contains 13 numeric features and 6 string features
* Case_No is an unwanted variable and hence can be removed from the dataset
* Based on the data dictionary Qchat_10_score column can also be dropped, however let us check the correlation with target variable and then decide upon it

In [39]:
# Dropping the unwanted variable

df.drop('Case_No', inplace=True, axis=1)

In [40]:
# Displaying first 5 rows

df.head()

A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Age_In_Months  Qchat_10_Score Sex  \
0   0   0   0   0   0   0   1   1   0    1             28               3   f   
1   1   1   0   0   0   1   1   0   0    0             36               4   m   
2   1   0   0   0   0   0   1   1   0    1             36               4   m   
3   1   1   1   1   1   1   1   1   1    1             24              10   m   
4   1   1   0   1   1   1   1   1   1    1             20               9   f   

        Ethnicity Jaundice Family_Mem_With_ASD Who_Completed_The_Test  \
0  middle eastern      yes                  no          family member   
1  White European      yes                  no          family member   
2  middle eastern      yes                  no          family member   
3        Hispanic       no                  no          family member   
4  White European       no                 yes          family member   

  Class_ASDTraits  
0              No  
1             Yes  
2             Yes  
3             Yes  
4             Yes

In [41]:
# Displaying last 5 rows

df.tail()

A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Age_In_Months  Qchat_10_Score  \
1049   0   0   0   0   0   0   0   0   0    1             24               1   
1050   0   0   1   1   1   0   1   0   1    0             12               5   
1051   1   0   1   1   1   1   1   1   1    1             18               9   
1052   1   0   0   0   0   0   0   1   0    1             19               3   
1053   1   1   0   0   1   1   0   1   1    0             24               6   

     Sex       Ethnicity Jaundice Family_Mem_With_ASD Who_Completed_The_Test  \
1049   f  White European       no                 yes          family member   
1050   m           black      yes                  no          family member   
1051   m  middle eastern      yes                  no          family member   
1052   m  White European       no                 yes          family member   
1053   m           asian      yes                 yes          family member   

     Class_ASDTraits  
1049              No  
1050             Yes  
1051             Yes  
1052              No  
1053             Yes

In [43]:
# descriptive summary of the numeric features

df.describe().T

count       mean       std   min   25%   50%   75%   max
A1              1054.0   0.563567  0.496178   0.0   0.0   1.0   1.0   1.0
A2              1054.0   0.448767  0.497604   0.0   0.0   0.0   1.0   1.0
A3              1054.0   0.401328  0.490400   0.0   0.0   0.0   1.0   1.0
A4              1054.0   0.512334  0.500085   0.0   0.0   1.0   1.0   1.0
A5              1054.0   0.524668  0.499628   0.0   0.0   1.0   1.0   1.0
A6              1054.0   0.576850  0.494293   0.0   0.0   1.0   1.0   1.0
A7              1054.0   0.649905  0.477226   0.0   0.0   1.0   1.0   1.0
A8              1054.0   0.459203  0.498569   0.0   0.0   0.0   1.0   1.0
A9              1054.0   0.489564  0.500128   0.0   0.0   0.0   1.0   1.0
A10             1054.0   0.586338  0.492723   0.0   0.0   1.0   1.0   1.0
Age_In_Months   1054.0  27.867173  7.980354  12.0  23.0  30.0  36.0  36.0
Qchat_10_Score  1054.0   5.212524  2.907304   0.0   3.0   5.0   8.0  10.0

#### Inference

* A1 to A10 can be considered as binary categorical variables
* Average age of a toddler is 27 months and w.r.t median age we can see the data is slightly left skewed
* The age ranges from 1 to 3 years
* Nearly 75% of the Qchat scores are above 3 which is the cut-off for ASD classification

In [44]:
# descriptive summary of the string features

df.describe(include='object').T

count unique             top  freq
Sex                     1054      2               m   735
Ethnicity               1054     11  White European   334
Jaundice                1054      2              no   766
Family_Mem_With_ASD     1054      2              no   884
Who_Completed_The_Test  1054      5   family member  1018
Class_ASDTraits         1054      2             Yes   728

#### Inference

* Male toddlers are found to be having the highest proportion
* Majority of the toddlers belong to white european ethinicity
* Most of the toddlers were not diagnosed with Jaundice at early stages
* More than 90% of the test givers were family members

In [51]:
# check for duplicate columns

print("Number of duplicate rows:", df.duplicated().sum())
df[df.duplicated()]

Number of duplicate rows: 79


A1  A2  A3  A4  A5  A6  A7  A8  A9  A10  Age_In_Months  Qchat_10_Score  \
33     0   0   0   0   0   0   0   0   0    0             12               0   
38     0   0   0   0   0   0   0   0   0    0             15               0   
39     0   0   0   0   0   0   0   0   0    1             15               1   
40     0   0   0   0   0   0   0   0   0    1             12               1   
42     0   0   0   0   0   0   0   0   0    0             15               0   
...   ..  ..  ..  ..  ..  ..  ..  ..  ..  ...            ...             ...   
968    0   0   0   0   1   1   0   1   0    1             35               4   
1006   1   0   1   1   1   0   1   0   0    1             24               6   
1022   0   0   0   0   0   0   0   0   0    0             36               0   
1026   0   0   0   0   0   0   0   0   0    1             36               1   
1044   1   1   1   1   1   1   1   1   1    1             25              10   

     Sex       Ethnicity Jaundice Family_Mem_With_ASD Who_Completed_The_Test  \
33     f  middle eastern       no                  no          family member   
38     f  middle eastern       no                  no          family member   
39     f  middle eastern       no                  no          family member   
40     f  middle eastern       no                  no          family member   
42     f           black       no                  no          family member   
...   ..             ...      ...                 ...                    ...   
968    f           asian       no                  no          family member   
1006   m           asian       no                  no          family member   
1022   f  White European       no                 yes          family member   
1026   m  White European       no                  no          family member   
1044   m           asian       no                 yes          family member   

     Class_ASDTraits  
33                No  
38                No  
39                No  
40                No  
42                No  
...              ...  
968              Yes  
1006             Yes  
1022              No  
1026              No  
1044             Yes  

[79 rows x 18 columns]

In [52]:
print("Unqiue values per column:")
df.nunique()

Unqiue values per column:


A1                         2
A2                         2
A3                         2
A4                         2
A5                         2
A6                         2
A7                         2
A8                         2
A9                         2
A10                        2
Age_In_Months             25
Qchat_10_Score            11
Sex                        2
Ethnicity                 11
Jaundice                   2
Family_Mem_With_ASD        2
Who_Completed_The_Test     5
Class_ASDTraits            2
dtype: int64